langchain/src/vectorstores/singlestore.ts

```diff
@@ -4,18 +4,79 @@ import type {
   OkPacket,
   ResultSetHeader,
   FieldPacket,
+  PoolOptions,
 } from "mysql2/promise";
 import { format } from "mysql2";
+import { createPool } from "mysql2/promise";
+import { Metadata } from "@opensearch-project/opensearch/api/types.js";
 import { VectorStore } from "./base.js";
 import { Embeddings } from "../embeddings/base.js";
 import { Document } from "../document.js";
 
-export interface SingleStoreVectorStoreConfig {
-  connectionPool: Pool;
+export enum DistanceMetrics {
```

Comment:
Using an enum will unfortunately break builds for other non-Node envs due to how they get transpiled - removing should fix the build.